specification/cognitiveservices/data-plane/ComputerVision/preview/v3.1-preview.2/Ocr.json

```diff
@@ -0,0 +1,502 @@
+      "description": "Analyze batch operation result.",
+      "type": "object",
+      "required": [
+        "version",
+        "readResults"
+      ],
+      "properties": {
+        "version": {
+          "description": "Version of schema used for this result.",
+          "type": "string"
+        },
+        "readResults": {
+          "description": "Text extracted from the input.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ReadResult"
+          }
+        }
+      }
+    },
+    "Line": {
+      "description": "An object representing a recognized text line.",
+      "type": "object",
+      "required": [
+        "boundingBox",
+        "appearance",
+        "text",
+        "words"
+      ],
+      "properties": {
+        "language": {
+          "description": "The BCP-47 language code of the recognized text line. Only provided where the language of the line differs from the page's.",
+          "type": "string"
+        },
+        "boundingBox": {
+          "description": "Bounding box of a recognized line.",
+          "$ref": "#/definitions/BoundingBox"
+        },
+        "appearance": {
+          "description": "Style and styleConfidence of the text line.",
+          "type": "object",
+          "$ref": "#/definitions/Appearance"
+        },
+        "text": {
+          "description": "The text content of the line.",
+          "type": "string"
+        },
+        "words": {
+          "description": "List of words in the text line.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Word"
+          }
+        }
+      }
+    },
+    "Word": {
+      "description": "An object representing a recognized word.",
+      "type": "object",
+      "required": [
+        "boundingBox",
+        "text",
+        "confidence"
+      ],
+      "properties": {
+        "boundingBox": {
+          "description": "Bounding box of a recognized word.",
+          "$ref": "#/definitions/BoundingBox"
+        },
+        "text": {
+          "description": "The text content of the word.",
+          "type": "string"
+        },
+        "confidence": {
+          "description": "Qualitative confidence measure.",
+          "type": "number",
+          "format": "float"
+        }
+      }
+    },
+    "BoundingBox": {
+      "description": "Quadrangle bounding box, with coordinates in original image. The eight numbers represent the four points (x-coordinate, y-coordinate from the left-top corner of the image) of the detected rectangle from the left-top corner in the clockwise direction. For images, coordinates are in pixels. For PDF, coordinates are in inches.",
+      "type": "array",
+      "items": {
+        "type": "number",
+        "x-nullable": false
+      }
+    },
+    "Appearance": {
+      "description": "An object representing the style and styleConfidence.",
+      "type": "object",
+      "required": [
+        "style",
```

Comment:
Is `style` still required (after you dropped `null` as a valid value). I'd have to go back, but I believe that there were scenarios where you couldn't decide (e.g. script languages) - in which case the service would either return an explicit "unknown" or omit the key/value pair *or* the property would come back with print or handwriting and a 0 confidence

Reply:
Yes, when we couldn't return appearance in Chinese text lines, we should omit the key/value pair which means omitting the whole appearance object. I have remove the appearance in required filed.